woocommerce/payment-gateway/class-sv-wc-payment-gateway.php

```diff
@@ -4081,22 +4081,34 @@ public function remove_support( $feature ) {
 			$feature = array( $feature );
 		}
 
+		$did_remove = false; 
+
 		foreach ( $feature as $name ) {
 
-			unset( $this->supports[ array_search( $name, $this->supports ) ] );
+			// Bail if the feature already isn't supported
+			if( $key = array_search( $name, $this->supports ) !== false ){
 
-			/**
-			 * Payment Gateway Remove Support Action.
-			 *
-			 * Fired when removing support for a specific gateway feature. Allows other actors
-			 * (including ourselves) to take action when support is removed.
-			 *
-			 * @since 4.1.0
-			 *
-			 * @param SV_WC_Payment_Gateway $this instance
-			 * @param string $name of supported feature being removed
-			 */
-			do_action( 'wc_payment_gateway_' . $this->get_id() . '_removed_support_' . str_replace( '-', '_', $name ), $this, $name );
+				unset( $this->supports[ $key ] );
+				$did_remove = true;
+
+				/**
+				 * Payment Gateway Remove Support Action.
+				 *
+				 * Fired when removing support for a specific gateway feature. Allows other actors
+				 * (including ourselves) to take action when support is removed.
+				 *
+				 * @since 4.1.0
+				 *
+				 * @param SV_WC_Payment_Gateway $this instance
+				 * @param string $name of supported feature being removed
+				 */
+				do_action( 'wc_payment_gateway_' . $this->get_id() . '_removed_support_' . str_replace( '-', '_', $name ), $this, $name );
+			}
+		}
+
+		if( $did_remove ){
+			// re-index to clean up any new gaps
+			$this->supports = array_values( $this->supports );
 		}
```

Comment:
no need this condition, we can always re-index

note the spacing with parenthesis is off (both in WP, SkyVerge or PSR they require a space)